libbpfgo/helpers/rwArray.go

```diff
@@ -0,0 +1,74 @@
+package helpers
+
+import (
+	"sync"
+)
+
+type slot struct {
+	value interface{}
+	used  bool
+}
+
+// RWArray allows for multiple concurrent readers but
+// only a single writer. The writers lock a mutex while the readers
+// are lock free.
+// It is implemented as an array of slots where each slot holds a
+// value (of type interface{}) and a boolean marker to indicate if it's
+// in use or not. The insertion (Put) performs a linear probe
+// looking for an available slot as indicated by the in-use marker.
+// While probing, it is not touching the value itself, as it's
+// being read without a lock by the readers.
+type RWArray struct {
+	slots []slot
+	mux   sync.Mutex
```

Comment:
nit: Conventionally mutexes are named as `mu`. Routers are `mux`.
```suggestion
	mu   sync.Mutex
```

Reply:
Good point. C++ habit 😀